src/speed_challenge_ctrl.cpp

```diff
@@ -32,14 +50,211 @@ class SpeedChallenge{
 
     ros::NodeHandle nh_;
     ros::NodeHandle private_nh_;
-    ros::Subscriber prop_sub;
 
-    prop_mapper::PropArray prop_array_test;
-    prop_mapper::PropArray prop_array;
+    ros::Subscriber prop_sub_;
+    ros::Subscriber task_to_exec_;
+    ros::Subscriber global_pos_;
+    ros::Publisher task_pos_;
+   
+    //Params 
+    float avoidance_angle;
+    float avoidance_distance;
+    float acceptable_error;
+
+    std::string local_pose_topic_;
+
+    geometry_msgs::PoseStamped current_pos_;
+
+    // Direction vectors
+    geometry_msgs::Point starting_pos_2_prop;
+    geometry_msgs::Point point1_direction_vec;
+    geometry_msgs::Point point2_direction_vec;
+    geometry_msgs::Point point3_direction_vec;
+    
+    geometry_msgs::PoseStamped starting_pos;
+
+    enum states {NOT_STARTED, MOVE_TO_POINT1, MOVE_TO_POINT2, MOVE_TO_POINT3, RETURN_TO_START};
+
+    states status = states::NOT_STARTED;
+
+    prop_mapper::PropArray prop_array_;
+
+    prop_mapper::Prop yellow_prop;
+
+    task_master::TaskGoalPosition task_goal_pos;
+
+    bool isReached() {
+
+        bool atDestination = false;
+
+        // TODO: use conversion node 
+        // Converting to gazebo coordinates 
+        float current_pos_x = current_pos_.pose.position.y;
+        float current_pos_y = -1 * current_pos_.pose.position.x;
+
+        if (current_pos_x < task_goal_pos.point.x + acceptable_error & current_pos_x > task_goal_pos.point.x - acceptable_error) {
+            if (current_pos_y < task_goal_pos.point.y + acceptable_error & current_pos_y > task_goal_pos.point.y - acceptable_error) {
+                atDestination = true;
+            }
+        }
+
+        return atDestination;
+    }
 
+    // Get prop array 
     void propCallback(const prop_mapper::PropArray msg)
     {
-        prop_array_test = msg;
+        prop_array_ = msg;
+    }
+
+    void taskCallback(const task_master::Task msg){
+
+        find_yellow_prop();
```

Comment:
can we remove yellow_prop as a global variable and instantiate it here? Could make find_yellow_prop() return a boolean for if the prop was found or not, and then pass in a null prop by reference.